--- a/rules/60-persistent-storage.rules	2013-05-19 22:52:03.694014013 -0500
+++ b/rules/60-persistent-storage.rules	2013-05-19 22:53:04.483233585 -0500
@@ -69,7 +69,7 @@
   IMPORT{builtin}="blkid --noraid"
 
 # probe filesystem metadata of disks
-KERNEL!="sr*", IMPORT{builtin}="blkid"
+KERNEL!="sr*|mmcblk0rpmb", IMPORT{builtin}="blkid"
 
 # watch metadata changes by tools closing the device after writing
 KERNEL!="sr*", OPTIONS+="watch"
