--- tcf-qoriqdbg-4.0.0.105390-r2/tcf-qoriqdbg-4.0.0.105390/core.internal/tcpTransport.cpp	2012-04-05 22:16:07.000000000 -0500
+++ tcf-qoriqdbg-4.0.0.105390/core.internal/tcpTransport.cpp	2012-10-30 20:42:11.648654005 -0500
@@ -35,7 +35,7 @@
 #include "channelImpl.h"
 #include "socketSupport.h"
 #include "socketTrace.h"
-
+#include <unistd.h> 
 #include <sstream>
 #include <iomanip>
 #include <cerrno>
