diff -urN cantest-1.0/cansend.c cantest-1.0-new/cansend.c
--- cantest-1.0/cansend.c	2008-09-27 16:30:01.000000000 +0800
+++ cantest-1.0-new/cansend.c	2008-09-27 16:29:02.000000000 +0800
@@ -63,7 +63,7 @@
 int main(int argc, char **argv)
 {
     int s; /* can raw socket */ 
-    int nbytes;
+    int nbytes, ret;
     struct sockaddr_can addr;
     struct can_frame frame, rframe;
     struct ifreq ifr;
@@ -100,7 +100,13 @@
     addr.can_ifindex = ifr.ifr_ifindex;
 
     ifr.ifr_ifru.ifru_ivalue = 1000000/2;
-    ioctl(s, SIOCSCANBAUDRATE, &ifr);
+    ret = ioctl(s, SIOCSCANBAUDRATE, &ifr);
+    if (ret) {
+	    printf("Err: cantest failed to set up baud rate for can port %s\n",
+			    					argv[1]);
+	    printf("     Please check the settings of system clock.\n");
+	    exit(1);
+    }
 
     if (bind(s, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
       perror("bind");
