diff -urN cantest-1.0.orig/Makefile cantest-1.0/Makefile
--- cantest-1.0.orig/Makefile	Thu Jan  1 08:00:00 1970
+++ cantest-1.0/Makefile	Fri May 16 08:46:06 2008
@@ -0,0 +1,15 @@
+
+
+EXEC = cantest
+OBJS = cansend.o lib.o
+
+all: $(EXEC)
+
+$(EXEC): $(OBJS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LDLIBS)
+
+romfs:
+	$(ROMFSINST) -e CONFIG_USER_CAN_TEST /usr/bin/cantest
+
+clean:
+	-rm -f $(EXEC) *.elf *.gdb *.o
diff -urN cantest-1.0.orig/cansend.c cantest-1.0/cansend.c
--- cantest-1.0.orig/cansend.c	Fri May 16 08:42:46 2008
+++ cantest-1.0/cansend.c	Fri May 16 08:59:05 2008
@@ -53,17 +53,19 @@
 #include <net/if.h>
 #include <sys/ioctl.h>
 
-#include <linux/can.h>
-#include <linux/can/raw.h>
+#include "can.h"
+#include "raw.h"
 
 #include "lib.h"
 
+#define SIOCSCANBAUDRATE	0x89F0
+
 int main(int argc, char **argv)
 {
     int s; /* can raw socket */ 
     int nbytes;
     struct sockaddr_can addr;
-    struct can_frame frame;
+    struct can_frame frame, rframe;
     struct ifreq ifr;
 
     /* check command line options */
@@ -91,20 +93,14 @@
       return 1;
     }
 
-    addr.can_family = AF_CAN;
-
     strcpy(ifr.ifr_name, argv[1]);
-    if (ioctl(s, SIOCGIFINDEX, &ifr) < 0) {
-      perror("SIOCGIFINDEX");
-      return 1;
-    }
+    ioctl(s, SIOCGIFINDEX, &ifr);
+
+    addr.can_family = AF_CAN;
     addr.can_ifindex = ifr.ifr_ifindex;
 
-    /* disable default receive filter on this RAW socket */
-    /* This is obsolete as we do not read from the socket at all, but for */
-    /* this reason we can remove the receive list in the Kernel to save a */
-    /* little (really a very little!) CPU usage.                          */
-    setsockopt(s, SOL_CAN_RAW, CAN_RAW_FILTER, NULL, 0);
+    ifr.ifr_ifru.ifru_ivalue = 1000000/2;
+    ioctl(s, SIOCSCANBAUDRATE, &ifr);
 
     if (bind(s, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
       perror("bind");
@@ -117,7 +113,14 @@
       return 1;
     }
 
-    //fprint_long_canframe(stdout, &frame, "\n", 0);
+    if ((nbytes = read(s, &rframe, sizeof(rframe))) < 0) {
+        perror("rx frame read");
+        return 0; /* quit */
+    }
+
+    printf("\nread %d bytes\n", nbytes);
+
+    fprint_long_canframe(stdout, &rframe, "\n", 0);
 
     close(s);
 
diff -urN cantest-1.0.orig/lib.c cantest-1.0/lib.c
--- cantest-1.0.orig/lib.c	Fri May 16 08:43:04 2008
+++ cantest-1.0/lib.c	Fri May 16 08:53:59 2008
@@ -49,8 +49,8 @@
 #include <string.h>
 
 #include <sys/socket.h> /* for sa_family_t */
-#include <linux/can.h>
 
+#include "can.h"
 #include "lib.h"
 
 #define CANID_DELIM '#'
diff -urN cantest-1.0.orig/raw.h cantest-1.0/raw.h
--- cantest-1.0.orig/raw.h	Fri May 16 08:44:58 2008
+++ cantest-1.0/raw.h	Wed Jan  9 15:58:37 2008
@@ -17,7 +17,10 @@
 #ifndef CAN_RAW_H
 #define CAN_RAW_H
 
-#include <linux/can.h>
+#include "can.h"
+
+#define PF_CAN		AF_CAN
+#define AF_CAN		29	/* Controller Area Network      */
 
 #define SOL_CAN_RAW (SOL_CAN_BASE + CAN_RAW)
 
