--- a/init.org	2013-03-14 05:44:19.000000000 -0500
+++ a/init	2013-03-14 05:44:47.000000000 -0500
@@ -35,10 +35,6 @@
     # propagate /dev from /sys
     echo "Starting udev"
 
-    # mount the tmpfs on /dev, if not already done
-    LANG=C awk '$2 == "/dev" && ($3 == "tmpfs" || $3 == "devtmpfs") { exit 1 }' /proc/mounts && {
-            mount -n -o mode=0755 -t tmpfs none "/dev"
-    }
     [ -e /dev/pts ] || mkdir -m 0755 /dev/pts
     [ -e /dev/shm ] || mkdir -m 1777 /dev/shm
     mount -a -t tmpfs 2>/dev/null
